src/hasReachedStyle.js

```diff
@@ -1,4 +1,6 @@
-export default function hasReachedStyle(currentStyle, style) {
+/* @flow */
+export default function hasReachedStyle(currentStyle: Object,
```

Comment:
1. I don’t understand why it isn’t cross-platform. Because you would need to hardcode all the possible keys? But then we would have to track all possible CSS properties. I do not think we should enforce the validity of the CSS property at the type level.
2. I prefer this solution. But couldn’t `randomKey` become an extra parameter instead of living inside the same configuration object. We would have a config that includes the essentials that every animation needs, and could be precisley type-checked and a separate object for random stuff. Right now the situation is a bit contradictory as the code operates on the style objects with functions that are extremely generic and would almost work for any type of object, yet we also wish to impose constraints on those same objects. 'Isolate the parts that change from those that don’t' might also apply to types.